docker-entrypoint.sh

```diff
@@ -19,6 +19,33 @@ run_as() {
     fi
 }
 
+# Execute all executable files in a given directory in alphanumeric order
+run_path() {
+    local hook_folder_path="/docker-entrypoint-hooks.d/$1"
+    local return_code=0
+
+    echo "Searching for scripts to run in the folder: ${hook_folder_path}"
+
+    (
+        cd "${hook_folder_path}"
+        find . -type f -print | while read script_file_path; do
+            if ! [ -x "${script_file_path}" ] && [ -f "${script_file_path}" ]; then
```

Comment:
The "strong reason" for allowing anything executable was to have a way to make an easy to customizable setup in your Nextcloud setup.
Without having to make pull request for every single feature which you would like to added to nextcloud-docker and at the same time lower the complexity.

This way you can have a script there install apps the first time you start your docker container and have another script which configure these apps.
One example could be Onlyoffice, you only need to install and configure it the first time you start Nextcloud and after that it is not needed again.

And to your example. You could have a script which  would do something to the postgres database and there you go magic, no need for any new pull request.
Just in general it would make having a custom test environment, for test if something works or not much easier, and at the end, you could just blow everything away again. No need to for manually configurations of anything.

Also, if you don't volume mount a folder into the container with any executable files it will not run any extra files, just the standard stuff.

Hope this answered your question 🙂

Reply:
aaarh, now I understand what you are asking :sweat_smile: Well, my logic was that if someone wanted to run some static binary file, or maybe php-scripts they still would be able 🙂
But if that is what they want, they can just use a shell script to start it (if they want to try to go that way), so if you think it will make it easier to maintain by having it be shell scripts, I am happy to make that change :grin:

Reply:
Hey @J0WI, I have now done so that only files ending in `.sh` and have the executable flag, will be executed.
I also, did so that the files are sorted before execution because I realized that I had forgotten to do that :sweat_smile:

Reply:
I will consider this resolved, unless there are more comments 😉